community/values/src/main/java/org/neo4j/values/storable/NumberValues.java

```diff
@@ -49,53 +67,110 @@ public static int hash( double number )
             return hash( asLong );
         }
         long bits = Double.doubleToLongBits( number );
-        return (int)(bits ^ (bits >>> 32));
+        return (int) (bits ^ (bits >>> 32));
     }
 
+    /*
+     * This is a slightly silly optimization but by turning the computation
+     * of the hashcode into a dot product we trick the jit compiler to use SIMD
+     * instructions and performance doubles.
+     */
     public static int hash( byte[] values )
     {
-        int result = 1;
-        for ( byte value : values )
+        final int max = values.length;
+        int result = COEFFICIENTS[max];
```

Comment:
What if `values.length` is longer than `COEFFICIENTS.length`?

Reply:
Hmm...I just changed that. But you're of course right, then catastrophe entails...